Validação da documentação do Gateway HTTP (#80)

robertatakenaka opened the pull request:
New fields for AttemptsAPI: checkin and validations which are the checkpoints which contains notices.
Get a single ticket (it was missing since the first version of the documentation).
Ticket and Comment have new structure: Ticket has author and title, and Comment has author and date.
Fixed payload structure.

robertatakenaka committed:
Adjustments in the documentation of Gateway HTTP

robertatakenaka committed:
Rewritting Gateway HTTP documentation

New fields for AttemptsAPI: checkin and validations which are the checkpoints which contains notices.
Get a single ticket (it was missing since the first version of the documentation).
Ticket and Comment have new structure: Ticket has author and title, and Comment has author and date.
Fixed payload structure.
~

gustavofonseca commented on docs/dev/attempts-api.rst:
```diff
@@ -39,32 +39,88 @@ Optional Parameters:
+            {
+              "label": "Checkin",
+              "message": "",
+              "status": 1,
```
`status` poderia mostrar a representação textual, exemplo `status.name`

robertatakenaka replied:
corrigido

Roberta Takenaka

SciELO 15 anos http://www.scielo15.org
conferencistashttp://www.scielo15.org/conferencistas/
 programa http://www.scielo15.org/programa/

SciELO - Scientific Electronic Library Online
FAPESP - CNPq - BIREME - FapUNIFESP
www.scielo.org | www.scielo.br

---

Avenida Onze de Junho, 269
04041-050 - São Paulo-SP - Brasil
Tel.: 55-11-5083-3639

On Tue, Sep 24, 2013 at 11:19 AM, Gustavo Fonseca
notifications@github.comwrote:

> In docs/dev/attempts-api.rst:
> 
> > -        "id": 1,
> > -        "collection_uri": "/api/v1/collections/1/",
> > -        "finished_at": "2012-07-24T21:59:23.909404",
> > -        "filepath": "/files/ajfajfajfa",
> > -        "is_valid": true,
> > -        "package_checksum": "12345678901234567890123456789012",
> > -        "resource_uri": "/api/v1/attempts/1/",
> > -        "started_at": "2012-07-24T21:53:23.909404",
> > -        "checkin": {
> > -          "finished_at": "2012-07-24T21:53:23.909404",
> > -          "start_at": "2012-07-24T21:53:23.909404",
> > -          "notices": [
> > -            {
> > -              "label": "Checkin",
> > -              "message": "",
> > -              "status": 1,
> 
> status poderia mostrar a representação textual, exemplo status.name
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/scieloorg/balaio/pull/128/files#r6544549
> .

## 

 http://www.scielo15.org/
*
*
_SciELO 15 anos_
http://www.scielo15.org

gustavofonseca commented on docs/dev/tickets-api.rst:
```diff
@@ -6,58 +6,101 @@ List all tickets
-  ]
+    "meta": {
+      "limit": 20,
+      "next": "/api/v1/packages/?limit=20&offset=40",
```
/api/v1/tickets/

robertatakenaka replied:
corrigido

Roberta Takenaka

SciELO 15 anos http://www.scielo15.org
conferencistashttp://www.scielo15.org/conferencistas/
 programa http://www.scielo15.org/programa/

SciELO - Scientific Electronic Library Online
FAPESP - CNPq - BIREME - FapUNIFESP
www.scielo.org | www.scielo.br

---

Avenida Onze de Junho, 269
04041-050 - São Paulo-SP - Brasil
Tel.: 55-11-5083-3639

On Tue, Sep 24, 2013 at 11:51 AM, Gustavo Fonseca
notifications@github.comwrote:

> In docs/dev/tickets-api.rst:
> 
> > -      "id": "1",
> > -      "finished_at": "2012-07-24T21:59:23.909404",
> > -      "is_open": true,
> > -      "resource_uri": "/api/v1/tickets/1/",
> > -      "started_at": "2012-07-24T21:53:23.909404",
> > -      "comments": [
> > -        "comments ...",
> > -        "comments ...",
> > -        "comments ... ",
> > 
> > ## \-      ],
> > -    }
> > -  ]
> > -    "meta": {
> > -      "limit": 20,
> > -      "next": "/api/v1/packages/?limit=20&offset=40",
> 
> /api/v1/tickets/
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/scieloorg/balaio/pull/128/files#r6545456
> .

## 

 http://www.scielo15.org/
*
*
_SciELO 15 anos_
http://www.scielo15.org

gustavofonseca commented on docs/dev/tickets-api.rst:
```diff
@@ -6,58 +6,101 @@ List all tickets
+        "title": "título para o ticket",
+        "comments": [
+          {
+            "comment_author": "user.name@scielo.org",
```
esse prefixo `comment` é meio desnecessário

robertatakenaka replied:
corrigido

Roberta Takenaka

SciELO 15 anos http://www.scielo15.org
conferencistashttp://www.scielo15.org/conferencistas/
 programa http://www.scielo15.org/programa/

SciELO - Scientific Electronic Library Online
FAPESP - CNPq - BIREME - FapUNIFESP
www.scielo.org | www.scielo.br

---

Avenida Onze de Junho, 269
04041-050 - São Paulo-SP - Brasil
Tel.: 55-11-5083-3639

On Tue, Sep 24, 2013 at 11:52 AM, Gustavo Fonseca
notifications@github.comwrote:

> In docs/dev/tickets-api.rst:
> 
> > -      "previous": "/api/v1/packages/?limit=20&offset=0",
> > -      "total_count": 100
> > -    },
> > -    "objects": [
> > -      {
> > -        "articlepkg_id": 1,
> > -        "id": 1,
> > -        "finished_at": "2012-07-24T21:59:23.909404",
> > -        "is_open": true,
> > -        "resource_uri": "/api/v1/tickets/1/",
> > -        "started_at": "2012-07-24T21:53:23.909404",
> > -        "ticket_author": "username@scielo.org",
> > -        "title": "título para o ticket",
> > -        "comments": [
> > -          {
> > -            "comment_author": "user.name@scielo.org",
> 
> esse prefixo comment é meio desnecessário
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/scieloorg/balaio/pull/128/files#r6545498
> .

## 

 http://www.scielo15.org/
*
*
_SciELO 15 anos_
http://www.scielo15.org

robertatakenaka committed:
status in notice is String, instead of number

robertatakenaka committed:
Fixes the next and previous route for tickets and attempts
Remove ticket_ from ticket_author and comment_ from comment_author and comment_date.

robertatakenaka committed:
Merge branch 'master' of git://github.com/scieloorg/balaio into 80_doc_rev3

Conflicts:
	docs/dev/tickets-api.rst

robertatakenaka committed:
Fixes issue_suppl_volume and issue_suppl_number

fabiobatalha commented on docs/dev/attempts-api.rst:
```diff
@@ -39,32 +39,83 @@ Optional Parameters:
+            {
+              "label": "references",
+              "message": "",
+              "status": 1,
```
Os demais exemplos de status estão com **ok, error e warning**

O que seria o 1?

robertatakenaka replied:
está incorreto. teria que ser ou ok, ou error, ou warning.

fabiobatalha commented on docs/dev/attempts-api.rst:
```diff
@@ -39,32 +39,83 @@ Optional Parameters:
+        },
+        "validations": {
+          "finished_at": "2012-07-24T21:53:23.909404",
+          "start_at": "2012-07-24T21:53:23.909404",
```
Não entendi a função dos atributos **finished_at** e **start_at** no grupo de validations.

O que começou e terminou? o ciclo de validações? se é isso, esses atributos podem ser os que estão no nível dos atributos do attempt.

robertatakenaka replied:
validations e checkin são dois checkpoints, por isso, ambos tem data de entrada e data de saída.

fabiobatalha replied:
ok

fabiobatalha commented on docs/dev/attempts-api.rst:
```diff
@@ -39,32 +39,83 @@ Optional Parameters:
+        "started_at": "2012-07-24T21:53:23.909404",
+        "checkin": {
+          "finished_at": "2012-07-24T21:53:23.909404",
+          "start_at": "2012-07-24T21:53:23.909404",
```
start_at deveria ser nomeado no passado também, como foi feito no nível do attempt para manter a consistência dos nomes.

robertatakenaka replied:
Sim, com certeza.

Roberta Takenaka

SciELO 15 anos http://www.scielo15.org
conferencistashttp://www.scielo15.org/conferencistas/
 programa http://www.scielo15.org/programa/

SciELO - Scientific Electronic Library Online
FAPESP - CNPq - BIREME - FapUNIFESP
www.scielo.org | www.scielo.br

---

Avenida Onze de Junho, 269
04041-050 - São Paulo-SP - Brasil
Tel.: 55-11-5083-3639

On Thu, Sep 26, 2013 at 4:14 PM, Fabio Batalha notifications@github.comwrote:

> In docs/dev/attempts-api.rst:
> 
> > -      "total_count": 100
> > -    },
> > -    "objects": [
> > -      {
> > -        "articlepkg_id": 1,
> > -        "id": 1,
> > -        "collection_uri": "/api/v1/collections/1/",
> > -        "finished_at": "2012-07-24T21:59:23.909404",
> > -        "filepath": "/files/ajfajfajfa",
> > -        "is_valid": true,
> > -        "package_checksum": "12345678901234567890123456789012",
> > -        "resource_uri": "/api/v1/attempts/1/",
> > -        "started_at": "2012-07-24T21:53:23.909404",
> > -        "checkin": {
> > -          "finished_at": "2012-07-24T21:53:23.909404",
> > -          "start_at": "2012-07-24T21:53:23.909404",
> 
> start_at deveria ser nomeado no passado também, como foi feito no nível do
> attempt para manter a consistência dos nomes.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/scieloorg/balaio/pull/128/files#r6611171
> .

## 

 http://www.scielo15.org/
*
*
_SciELO 15 anos_
http://www.scielo15.org

fabiobatalha commented on docs/dev/attempts-api.rst:
```diff
@@ -88,18 +139,65 @@ Response::
-    
+    "checkin": {
+      "finished_at": "2012-07-24T21:53:23.909404",
+      "start_at": "2012-07-24T21:53:23.909404",
```
deixar started_at para manter padrão de nomenclatura.

fabiobatalha commented:
GET /api/v1/packages/
- Acho que deve ser incluído o ano de publicação para compor filtros por ano, volume e numero.

POST /api/v1/tickets/
- Acho que **message** não pode ser opcional.

PATCH /api/v1/tickets/:id/
- Acho que o **comment_author** e **message** não podem ser opcionais.

robertatakenaka committed:
Merge branch 'master' of git://github.com/scieloorg/balaio into 80_doc_rev3

robertatakenaka commented:
```
POST /api/v1/tickets/

Acho que message não pode ser opcional.
PATCH /api/v1/tickets/:id/

Acho que o comment_author e message não podem ser opcionais.

```

Podem ser opcionais, por exemplo, no caso de close do ticket. Quem fecha o ticket não precisaria fazer nenhum comentário. E aí? O que você sugere?

robertatakenaka committed:
Ajustes de acordo com os comentarios do revisor do pull

robertatakenaka committed:
voltei "collection_uri": "/api/v1/collections/1/",

fabiobatalha commented:
O problema de deixar mensagem opcional para todos as demais ações é que provavelmente nunca ninguém vai comentar nada. Mas podemos deixar aberto e avaliar o uso da ferramenta, se isso se tornar verdade podemos criar um endpoint apenas para encerrar o ticket.

robertatakenaka commented:
Sim, é muito provável que será criado o close.

Roberta Takenaka

SciELO 15 anos http://www.scielo15.org
conferencistashttp://www.scielo15.org/conferencistas/
 programa http://www.scielo15.org/programa/

SciELO - Scientific Electronic Library Online
FAPESP - CNPq - BIREME - FapUNIFESP
www.scielo.org | www.scielo.br

---

Avenida Onze de Junho, 269
04041-050 - São Paulo-SP - Brasil
Tel.: 55-11-5083-3639

2013/9/26 Fabio Batalha notifications@github.com

> O problema de deixar mensagem opcional para todos as demais ações é que
> provavelmente nunca ninguém vai comentar nada. Mas podemos deixar aberto e
> avaliar o uso da ferramenta, se isso se tornar verdade podemos criar um
> endpoint apenas para encerrar o ticket.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/scieloorg/balaio/pull/128#issuecomment-25199753
> .

## 

 http://www.scielo15.org/
*
*
_SciELO 15 anos_
http://www.scielo15.org

robertatakenaka committed:
Adicionando issue_year no filtro de packages

robertatakenaka commented:
Efetuados os ajustes recomendados. Por favor, reavaliar.
